crates/nu-color-config/src/shape_color.rs

```diff
@@ -3,143 +3,28 @@ use nu_ansi_term::{Color, Style};
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Blue)
-            }
-        }
-        "flatshape_operator" => {
-            if conf.color_config.contains_key("flatshape_operator") {
-                let int_color = &conf.color_config["flatshape_operator"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Yellow)
-            }
-        }
-        "flatshape_signature" => {
-            if conf.color_config.contains_key("flatshape_signature") {
-                let int_color = &conf.color_config["flatshape_signature"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Green).bold()
-            }
-        }
-        "flatshape_string" => {
-            if conf.color_config.contains_key("flatshape_string") {
-                let int_color = &conf.color_config["flatshape_string"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Green)
-            }
-        }
-        "flatshape_filepath" => {
-            if conf.color_config.contains_key("flatshape_filepath") {
-                let int_color = &conf.color_config["flatshape_filepath"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Cyan)
-            }
-        }
-        "flatshape_globpattern" => {
-            if conf.color_config.contains_key("flatshape_globpattern") {
-                let int_color = &conf.color_config["flatshape_globpattern"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Cyan).bold()
-            }
-        }
-        "flatshape_variable" => {
-            if conf.color_config.contains_key("flatshape_variable") {
-                let int_color = &conf.color_config["flatshape_variable"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Purple)
-            }
-        }
-        "flatshape_flag" => {
-            if conf.color_config.contains_key("flatshape_flag") {
-                let int_color = &conf.color_config["flatshape_flag"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Blue).bold()
-            }
-        }
-        "flatshape_custom" => {
-            if conf.color_config.contains_key("flatshape_custom") {
-                let int_color = &conf.color_config["flatshape_custom"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().bold()
-            }
-        }
-        _ => Style::default(),
+    match conf.color_config.get(shape.as_str()) {
```

Comment:
the reason i added `conf.color_config.contains_key["blah"]` is because `conf.color_config["blah"]` would puke if the key didn't exist. i guess `conf.color_config.get()` doesn't behave that way?

Reply:
It returns and option instead of error. That allowed me to clean most of the code

Reply:
ah, ok. I didn't realize it returned an option. nice.